pkgs/rocky.dockerfile

```diff
@@ -0,0 +1,99 @@
+# syntax=docker/dockerfile:1.5
+
+ARG REDHAT_VER=9
+
+FROM rockylinux:${REDHAT_VER} AS builder-rocky
+
+ARG ARCH=x86_64
+ARG POSTGRES_VER=16
+ARG REDHAT_VER
+
+# UPDATE
+RUN dnf update -y
+
+# SUDO IS USEFUL IF YOU WANT TO RUN AS YOUR OWN HOST'S USER & NOT ROOT
+# E.G. `docker run -i -t --rm --user $(id -u):100 -v $PWD:/omni <image>`
+# THIS WILL POP YOU INTO A CONTAINER WHERE YOU ARE 'redhat' IN THE GROUP
+# 'users'. AND THE FILES YOU TOUCH/WRITE IN /omni WILL BE OWNED BY THE SAME
+# USER ID AS YOUR HOST.
+RUN dnf install -y sudo
+RUN echo 'redhat ALL=(ALL) NOPASSWD: ALL' | tee /etc/sudoers.d/redhat
+RUN useradd -g users -m redhat
+
+# POSTGRES
+RUN dnf install -y \
+    https://download.postgresql.org/pub/repos/yum/reporpms/EL-${REDHAT_VER}-$(uname -m)/pgdg-redhat-repo-latest.noarch.rpm
+RUN dnf --enablerepo=crb install -y \
+    postgresql${POSTGRES_VER}-contrib \
+    postgresql${POSTGRES_VER}-devel \
+    postgresql${POSTGRES_VER}-plpython3 \
+    postgresql${POSTGRES_VER}-server
+
+# DEPS
+RUN yum groupinstall -y "Development Tools"
+RUN dnf --enablerepo=crb install -y \
+    cmake \
+    doxygen \
+    nc \
+    openssl-devel \
+    perl-CPAN \
+    python3-devel
+
+# FPM
+RUN dnf install -y ruby ruby-devel rubygems squashfs-tools
+ENV GEM_HOME=/usr/local
+RUN gem install fpm
+
+ARG ITERATION=1
+
+COPY ./ /omni
+WORKDIR /build
+
+# BUILD
+ENV PATH /usr/pgsql-${POSTGRES_VER}/bin:${PATH}
+RUN cmake -DPG_CONFIG=$(which pg_config) -DOPENSSL_CONFIGURED=1 /omni
+RUN make -j all
+RUN make package_extensions
+
+WORKDIR /pkgs
+
+# PACKAGING
+RUN \
+for pkg in $(cat /build/artifacts.txt|awk -F'=' '{print $1}') ; do \
+    pkgline=$(grep "^$pkg\=" /build/artifacts.txt) ;\
+    vers=$(echo $pkgline|cut -d'#' -f1|awk -F'=' '{print $2}') ;\
+    deps="--depends postgresql$POSTGRES_VER" ;\
+    for d in $(echo $pkgline|awk -F'#' '{print $2}'|tr ',' ' ') ; do \
+        n=$(echo $d|awk -F'=' '{print $1}') ;\
+        _=$(echo $d|awk -F'=' '{print $2}') ;\
+        case $n in \
+            dblink|pgcrypto) n="postgresql$POSTGRES_VER-contrib"   ;;\
+            plpy*)           n="postgresql$POSTGRES_VER-plpython3" ;;\
+        esac ;\
+        deps="$deps --depends $n" ;\
+    done ;\
+    tmp=$(mktemp -d) ;\
+    mkdir -p "$tmp/lib" ;\
+    mkdir -p "$tmp/share/extension" ;\
+    cp -a /build/packaged/$pkg--*.so        "$tmp/lib/"             2>/dev/null || true ;\
+    cp -a /build/packaged/extension/$pkg--* "$tmp/share/extension/" 2>/dev/null || true ;\
+    ${GEM_HOME}/bin/fpm \
+        --input-type dir \
+        --output-type rpm \
+        --name $pkg \
+        --version $vers \
+        --iteration $ITERATION \
+        --prefix /usr/pgsql-$POSTGRES_VER/ \
+        --chdir $tmp \
+        $deps \
+        . ;\
+done
+# NOTE THAT IT WOULD BE BETTER TO HAVE --depends 'xyz >= 1.0' FOR EACH
```

Comment:
I was able to solve it with the following patch:

```diff
diff --git a/pkgs/rocky.dockerfile b/pkgs/rocky.dockerfile
index d635082..6f3d461 100644
--- a/pkgs/rocky.dockerfile
+++ b/pkgs/rocky.dockerfile
@@ -65,19 +65,23 @@ for pkg in $(cat /build/artifacts.txt|awk -F'=' '{print $1}') ; do \
     deps="--depends postgresql$POSTGRES_VER" ;\
     for d in $(echo $pkgline|awk -F'#' '{print $2}'|tr ',' ' ') ; do \
         n=$(echo $d|awk -F'=' '{print $1}') ;\
-        _=$(echo $d|awk -F'=' '{print $2}') ;\
+        v=$(echo $d|awk -F'=' '{print $2}') ;\
         case $n in \
             dblink|pgcrypto) n="postgresql$POSTGRES_VER-contrib"   ;;\
             plpy*)           n="postgresql$POSTGRES_VER-plpython3" ;;\
         esac ;\
-        deps="$deps --depends $n" ;\
+        if [ "$v" = "*" ]; then \
+          deps="$deps --depends '$n'" ;\
+        else \
+          deps="$deps --depends '$n = $v'" ;\
+        fi \
     done ;\
     tmp=$(mktemp -d) ;\
     mkdir -p "$tmp/lib" ;\
     mkdir -p "$tmp/share/extension" ;\
     cp -a /build/packaged/$pkg--*.so        "$tmp/lib/"             2>/dev/null || true ;\
     cp -a /build/packaged/extension/$pkg--* "$tmp/share/extension/" 2>/dev/null || true ;\
-    ${GEM_HOME}/bin/fpm \
+    eval ${GEM_HOME}/bin/fpm \
         --input-type dir \
         --output-type rpm \
         --name $pkg \
```

Reply:
I had the if/case there to set the version of the dependency. It barfs on the build machine. Where are you trying this? This is not working on the latest fpm on rocky linux.

Reply:
This is the battle I had with fpm and the reason for the NOTE at the bottom.

Reply:
Oh maybe your eval is helping it behave. I'll give it a try.

Reply:
It absolutely did work here. Did you try applying the patch?

The crux of the solution is in adding `eval`

Reply:
Worked like this here:

```

#24 [18/19] RUN for pkg in $(cat /build/artifacts.txt|awk -F'=' '{print $1}') ; do     pkgline=$(grep "^$pkg\=" /build/artifacts.txt) ;    vers=$(echo $pkgline|cut -d'#' -f1|awk -F'=' '{print $2}') ;    deps="--depends postgresql16" ;    for d in $(echo $pkgline|awk -F'#' '{print $2}'|tr ',' ' ') ; do         n=$(echo $d|awk -F'=' '{print $1}') ;        v=$(echo $d|awk -F'=' '{print $2}') ;        case $n in             dblink|pgcrypto) n="postgresql16-contrib"   ;;            plpy*)           n="postgresql16-plpython3" ;;        esac ;        if [ "$v" = "*" ]; then           deps="$deps --depends '$n'" ;        else           deps="$deps --depends '$n = $v'" ;        fi     done ;    tmp=$(mktemp -d) ;    mkdir -p "$tmp/lib" ;    mkdir -p "$tmp/share/extension" ;    cp -a /build/packaged/$pkg--*.so        "$tmp/lib/"             2>/dev/null || true ;    cp -a /build/packaged/extension/$pkg--* "$tmp/share/extension/" 2>/dev/null || true ;    eval /usr/local/bin/fpm         --input-type dir         --output-type rpm         --name $pkg         --version $vers         --iteration 1         --prefix /usr/pgsql-16/         --chdir $tmp         $deps         . ;done
#24 0.284 {:timestamp=>"2024-08-21T15:37:08.411426+0000", :message=>"Created package", :path=>"omni-0.1.4-1.aarch64.rpm"}
#24 0.449 {:timestamp=>"2024-08-21T15:37:08.576818+0000", :message=>"Created package", :path=>"omni_http-0.1.0-1.aarch64.rpm"}
#24 0.637 {:timestamp=>"2024-08-21T15:37:08.765529+0000", :message=>"Created package", :path=>"omni_types-0.1.1-1.aarch64.rpm"}
#24 0.839 {:timestamp=>"2024-08-21T15:37:08.967073+0000", :message=>"Created package", :path=>"omni_sql-0.3.2-1.aarch64.rpm"}
#24 1.625 {:timestamp=>"2024-08-21T15:37:09.753040+0000", :message=>"Created package", :path=>"omni_httpc-0.1.1-1.aarch64.rpm"}
#24 1.837 {:timestamp=>"2024-08-21T15:37:09.965224+0000", :message=>"Created package", :path=>"omni_xml-0.1.1-1.aarch64.rpm"}
#24 2.019 {:timestamp=>"2024-08-21T15:37:10.146610+0000", :message=>"Created package", :path=>"omni_web-0.1.1-1.aarch64.rpm"}
#24 2.195 {:timestamp=>"2024-08-21T15:37:10.322648+0000", :message=>"Created package", :path=>"omni_aws-0.1.0-1.aarch64.rpm"}
#24 2.374 {:timestamp=>"2024-08-21T15:37:10.501933+0000", :message=>"Created package", :path=>"omni_containers-0.1.0-1.aarch64.rpm"}
#24 3.186 {:timestamp=>"2024-08-21T15:37:11.312926+0000", :message=>"Created package", :path=>"omni_httpd-0.1.2-1.aarch64.rpm"}
#24 3.351 {:timestamp=>"2024-08-21T15:37:11.478811+0000", :message=>"Created package", :path=>"omni_id-0.2.0-1.aarch64.rpm"}
#24 3.519 {:timestamp=>"2024-08-21T15:37:11.647603+0000", :message=>"Created package", :path=>"omni_json-0.1.0-1.aarch64.rpm"}
#24 3.685 {:timestamp=>"2024-08-21T15:37:11.813491+0000", :message=>"Created package", :path=>"omni_manifest-0.1.0-1.aarch64.rpm"}
#24 3.874 {:timestamp=>"2024-08-21T15:37:12.002646+0000", :message=>"Created package", :path=>"omni_mimetypes-0.1.0-1.aarch64.rpm"}
#24 4.044 {:timestamp=>"2024-08-21T15:37:12.172499+0000", :message=>"Created package", :path=>"omni_os-0.1.0-1.aarch64.rpm"}
#24 4.209 {:timestamp=>"2024-08-21T15:37:12.337573+0000", :message=>"Created package", :path=>"omni_python-0.1.0-1.aarch64.rpm"}
#24 4.376 {:timestamp=>"2024-08-21T15:37:12.504139+0000", :message=>"Created package", :path=>"omni_vfs_types_v1-0.1.0-1.aarch64.rpm"}
#24 4.552 {:timestamp=>"2024-08-21T15:37:12.679507+0000", :message=>"Created package", :path=>"omni_vfs-0.1.1-1.aarch64.rpm"}
#24 4.725 {:timestamp=>"2024-08-21T15:37:12.853684+0000", :message=>"Created package", :path=>"omni_schema-0.2.0-1.aarch64.rpm"}
#24 4.906 {:timestamp=>"2024-08-21T15:37:13.034665+0000", :message=>"Created package", :path=>"omni_seq-0.1.0-1.aarch64.rpm"}
#24 5.075 {:timestamp=>"2024-08-21T15:37:13.202571+0000", :message=>"Created package", :path=>"omni_var-0.2.1-1.aarch64.rpm"}
#24 5.244 {:timestamp=>"2024-08-21T15:37:13.372122+0000", :message=>"Created package", :path=>"omni_txn-0.2.0-1.aarch64.rpm"}
```

Reply:
This is the last thing I need to resolve. Trying again on the build machine.

Reply:
I think it's working now. Fingers crossed and knock on wood for one final run on the build box.